addressing/README.md

```diff
@@ -0,0 +1,329 @@
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+|-----------------|---------------|--------|-----------------|
+| 1A              | Working Draft | Active | r0, 2019-05-27  |
+
+
+Authors: [@yusefnapora]
+
+Interest Group: TBD
+
+[@yusefnapora]: https://github.com/yusefnapora
+
+See the [lifecycle document][lifecycle-spec] for context about maturity level
+and spec status.
+
+[lifecycle-spec]: https://github.com/libp2p/specs/blob/master/00-framework-01-spec-lifecycle.md
+
+## Table of Contents
+
+- [Addressing in libp2p](#addressing-in-libp2p)
+    - [Table of Contents](#table-of-contents)
+    - [Overview](#overview)
+    - [multiaddr in libp2p](#multiaddr-in-libp2p)
+        - [multiaddr basics](#multiaddr-basics)
+        - [Composing multiaddrs](#composing-multiaddrs)
+            - [Encapsulation](#encapsulation)
+            - [Decapsulation](#decapsulation)
+        - [The p2p multiaddr](#the-p2p-multiaddr)
+            - [Historical Note: the `ipfs` multiaddr Protocol](#historical-note-the-ipfs-multiaddr-protocol)
+    - [Transport multiaddrs](#transport-multiaddrs)
+        - [IP and Name Resolution](#ip-and-name-resolution)
+            - [dnsaddr Links](#dnsaddr-links)
+        - [TCP](#tcp)
+        - [WebSockets](#websockets)
+        - [QUIC](#quic)
+        - [`p2p-circuit` Relay Addresses](#p2p-circuit-relay-addresses)
+
+## Overview
+
+libp2p makes a distinction between a peer's **identity** and its **location**.
+A peer's identity is stable, verifiable, and valid for the entire lifetime of
+the peer (whatever that may be for a given application). Peer identities are
+derived from public keys as described in the [peer id spec][peer-id-spec].
+
+On a particular network, at a specific point in time, a peer may have one or
+more locations, which can be represented using addresses. For example, I may be
+reachable via the global IPv4 address of 7.7.7.7 on TCP port 1234.
+
+In a system that only supported TCP/IP or UDP over IP, we could easily write our
+addresses with the familiar `<ip>:<port>` notation and store them as tuples of
+address and port. However, libp2p was designed to be transport agnostic, which
+means that we can't assume that we'll even be using an IP-backed network at all.
+
+To support a growing set of transport protocols without special-casing each
+addressing scheme, libp2p uses [multiaddr][multiaddr-repo] to encode network
+addresses for all supported transport protocols.
+
+This document covers [how multiaddr is used in libp2p](#multiaddr-in-libp2p).
+For more information on other use cases, or to find links to multiaddr
+implementations in various languages, see the [mulitaddr
+repository][multiaddr-repo].
+
+## multiaddr in libp2p
+
+multiaddr is used throughout libp2p for encoding network addresses, and
+addresses are generally exchanged over the wire as binary-encoded multiaddrs in
+libp2p's core protocols. 
+
+When exchanging addresses, peers send a multiaddr containing both their network
```

Comment:
Revisiting this...

1. "network" address is usually referred to as the "transport" address (not critical, but it could involve relays and multiple network addresses).
2. We _usually_ send AddrInfos (`AddrInfo{ID: <peer-id>, Addrs: <transport-addrs>}`).

But this section is really fine as-is, this is just me nit picking.